VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java

```diff
@@ -987,53 +912,33 @@ public void onEvent(final NativeAd nativeAd, final String source, final String e
-            {
-                e( "Failed to register native ad view for interaction. Native ad is null" );
-                return;
-            }
-
-            final View.OnClickListener clickListener = new View.OnClickListener()
-            {
-                @Override
-                public void onClick(final View view)
-                {
-                    log( "Native ad clicked from click listener" );
+        public void prepareViewForInteraction(MaxNativeAdView maxNativeAdView) {
+            final NativeTextComponent titleComponent = (NativeTextComponent)nativeAd.getComponent("title");
+            final NativeTextComponent disclaimerComponent = (NativeTextComponent)nativeAd.getComponent("disclaimer");
+            final NativeTextComponent bodyComponent = (NativeTextComponent)nativeAd.getComponent("body");
+            final NativeTextComponent ctaComponent = (NativeTextComponent)nativeAd.getComponent("callToAction");
 
-                    nativeAd.invokeDefaultAction( view.getContext() );
-                    listener.onNativeAdClicked();
-                }
-            };
+            if (titleComponent != null && maxNativeAdView.getTitleTextView() != null) {
+                titleComponent.prepareView(maxNativeAdView.getTitleTextView());
+            }
+            if (disclaimerComponent != null && maxNativeAdView.getAdvertiserTextView() != null) {
+                disclaimerComponent.prepareView(maxNativeAdView.getAdvertiserTextView());
+            }
+            if (bodyComponent != null && maxNativeAdView.getBodyTextView() != null) {
+                bodyComponent.prepareView(maxNativeAdView.getBodyTextView());
+            }
+            if (ctaComponent != null && maxNativeAdView.getCallToActionButton() != null) {
+                ctaComponent.prepareView(maxNativeAdView.getCallToActionButton());
+            }
 
-            // Verizon's click registration methods don't work with views when recycling is involved
-            // so they told us to manually invoke it as AdMob does
-            AppLovinSdkUtils.runOnUiThread( new Runnable()
-            {
-                @Override
-                public void run()
-                {
-                    if ( nativeAdView.getTitleTextView() != null ) nativeAdView.getTitleTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getAdvertiserTextView() != null ) nativeAdView.getAdvertiserTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getBodyTextView() != null ) nativeAdView.getBodyTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getIconImageView() != null ) nativeAdView.getIconImageView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getCallToActionButton() != null ) nativeAdView.getCallToActionButton().setOnClickListener( clickListener );
-                }
-            } );
+            nativeAd.registerContainerView(maxNativeAdView);
```

Comment:
Could you check if multiple native ads using the same recycled `MaxNativeAdView` will allow clicks to work for all the views? We had this issue previously with Verizon's native ads so just want to make sure they fixed it